cypress/utils/apps/query_enhancements/commands.js

```diff
@@ -3,7 +3,87 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-Cypress.Commands.add('setQueryEditor', (value, opts = {}, submit = true) => {
+const MAX_RETRIES = 3;
+const RETRY_DELAY = 1000;
+
+const forceFocusEditor = () => {
+  return cy
+    .get('.globalQueryEditor .react-monaco-editor-container')
+    .click({ force: true })
+    .wait(200) // Give editor time to register focus
+    .get('.inputarea')
+    .focus()
+    .wait(200); // Wait for focus to take effect
+};
+
+const clearMonacoEditor = () => {
```

Comment:
One thing that usually works for me in Cypress is to use `.type("{selectall}{backspace}{selectall}{backspace}")` to clear inputs, not sure if it would work in this case though.

Reply:
thanks. so far this is the most stable way to avoid flakiness.